bindsnet/network/__init__.py

```diff
@@ -283,7 +293,13 @@ def run(self, inpts: Dict[str, torch.Tensor], time: int, **kwargs) -> None:
             for l in self.layers:
                 # Update each layer of nodes.
                 if isinstance(self.layers[l], AbstractInput):
-                    self.layers[l].forward(x=inpts[l][t])
+                    # grab the time slice for each input
```

Comment:
I'm not sure what these lines do, could you explain?

Reply:
Essentially, to deal with the fact that the time dimension could be anywhere I keep around a dictionary of slices for each input separately (different shape, time dimension, etc.) and I fill in the time into that list at the appropriate dimension. `slice(None)` is the same as `:`. I hope that helps clarify a bit.

Reply:
Yup, thanks.